internal/notification/ntfy.go

```diff
@@ -6,60 +6,85 @@ import (
 	"jetspotter/internal/jetspotter"
 )
 
+type NtfyAction struct {
+	Action string `json:"action"`
+	Clear  bool   `json:"clear,omitempty"`
+	Label  string `json:"label"`
+	URL    string `json:"url"`
+}
+
 type NtfyNotification struct {
-	Message  string   `json:"message,omitempty"`
-	Topic    string   `json:"topic,omitempty"`
-	Title    string   `json:"title,omitempty"`
-	Markdown bool     `json:"markdown,omitempty"`
-	Tags     []string `json:"tags,omitempty"`
+	Actions  []NtfyAction `json:"actions,omitempty"`
+	Message  string       `json:"message,omitempty"`
+	Topic    string       `json:"topic,omitempty"`
+	Title    string       `json:"title,omitempty"`
+	Markdown bool         `json:"markdown,omitempty"`
+	Tags     []string     `json:"tags,omitempty"`
 }
 
-// SendNtfyMessage sends a ntfy message containing metadata of a list of aircraft
+// SendNtfyMessage sends a ntfy message containing metadata of aircraft
+// Each aircraft will have its own separate notification
 func SendNtfyMessage(aircraft []jetspotter.AircraftOutput, config configuration.Config) error {
-	message, err := buildNtfyMessage(aircraft, config)
-	if err != nil {
-		return err
-	}
+	// Send a separate message for each aircraft
+	for _, ac := range aircraft {
+		// Build a message for a single aircraft
+		singleAircraftMessage, err := buildNtfyMessage([]jetspotter.AircraftOutput{ac}, config)
+		if (err != nil) {
+			return err
+		}
 
-	notification := Notification{
-		Message: message,
-		Type:    Ntfy,
-		URL:     config.NtfyServer,
-	}
+		notification := Notification{
+			Message: singleAircraftMessage,
+			Type:    Ntfy,
+			URL:     config.NtfyServer,
+		}
 
-	err = SendMessage(aircraft, notification)
-	if err != nil {
-		return err
+		err = SendMessage([]jetspotter.AircraftOutput{ac}, notification)
+		if err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
+// Constructor function for NtfyAction thats sets default value for Action and Clear
+func AddNtfyAction(label, url string) NtfyAction {
+	return NtfyAction{
+		Action: "view",
+		Clear:  true,
+		Label:  label,
+		URL:    url,
+	}
+}
+
 func buildNtfyMessage(aircraft []jetspotter.AircraftOutput, config configuration.Config) (message NtfyNotification, err error) {
 	message.Title = "An aircraft has been spotted!"
 	message.Topic = config.NtfyTopic
 	message.Tags = []string{"jetspotter"}
 	message.Markdown = true
 
 	for _, ac := range aircraft {
```

Comment:
Hey @vvanouytsel 
I beleive, this loop is redundant now. Since only one aircraft will be sent to `buildNtfyMessage` now.

Reply:
Indeed, should be fixed in https://github.com/vvanouytsel/jetspotter/commit/514890fca3ed978ec02297ffdb7e85b88c37507a.